### ISSUE TITLE
RL hyperparameter setting

### ISSUE THREAD

**Issue by Nipers** [PRIME-RL/PRIME#37]
I carefully compare the loss function in the notion pseudo code with the implementation code in the compute_rloo_returns function. Here is my question: 
According to the pseudo code, there is no GAE operation. in the outcome rewards calculation, GAE is only adopted when calculating process rewards.
But according to compute_rloo_returns, both of these two advantages are calculated with GAE.
I notice that different Gamma values are set for gt_scores and rm_scores, but in the provided training shell:
    algorithm.adv_params.verifier_gamma=1.0 \
    algorithm.adv_params.reward_model_gamma=1.0 \
 I wonder if this is the configuration for the model reported in the notion, i.e. Eurus-2-7B-PRIME, 
I might have  some misunderstanding about the implementation, looking forward to your feedback.

**Comment by ZefanW** [comment]
Hi! 
Our method does not use GAE or a critic model. Instead, only discounted rewards are used, with the process rewards being discounted and outcome rewards not being discounted. The pseudo-code correctly reflects this approach. 
Regarding the code configuration you mentioned, the purpose of having separate gamma values is to allow different gamma ( discount factor ) for the process reward and the outcome reward. When ``algorithm.adv_params_verifier_gamma`` is set to ``1.0``, the implementation aligns with the pseudo-code. 
Please feel free to share further thoughts!

**Comment by Nipers** [comment]
Thank you very much for your response, I might have figure out why I was confused.
In my original opinion, if you set algorithm.adv_params.verifier_gamma as 0, this means the sum up operation is not taken for the outcome reward, if it is set as 1, this means the sum up operation for the implicit reward is also taken for the outcome reward. This makes me confused. 
But now I realize that the reward_tensor for gt_score format is different with my original thought
I have read this:         
for i in range(verifier_reward.shape[0]):
            verifier_reward[i,valid_response_length[i]-1] += verifier_score[i]
        reward_tensor_dict['gt_scores'] = verifier_reward
        reward_metrics.update(verifier_metrics)
only the gt_score for the last token is set as 1 if the sample is correct. This makes everything reasonable.
